src/main/java/roomescape/auth/PermissionCheckInterceptor.java

```diff
@@ -0,0 +1,35 @@
+package roomescape.auth;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.lang.reflect.Method;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import roomescape.auth.exception.AuthenticationException;
+import roomescape.domain.role.MemberRole;
+import roomescape.domain.role.Role;
+
+public class PermissionCheckInterceptor implements HandlerInterceptor {
+    private final TokenManager tokenManager;
+
+    public PermissionCheckInterceptor(TokenManager tokenManager) {
+        this.tokenManager = tokenManager;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        HandlerMethod handlerMethod = (HandlerMethod) handler;
+        Method method = handlerMethod.getMethod();
+        PermissionRequired permissionRequired = method.getAnnotation(PermissionRequired.class);
+        if (permissionRequired == null) {
+            return true;
+        }
```

Comment:
```suggestion
        if (method.isAnnotationPresent(PermissionRequired.class) {
            return true;
        }
        PermissionRequired permissionRequired = method.getAnnotation(PermissionRequired.class);
```
구현의 방식에 따라 다르겠으나 null을 컨트롤하는 것보다는 해당 방법이 더 명확할 것 같네요.

Reply:
리플렉션 API 한번 알아봐야겠습니다 ㅎㅎ